### PR TITLE
Update supported versions to include Humble + Garden

### PR DESCRIPTION
**Pull request by j-rivero**
Update the supported version to include the combination of Humble + Garden from source.

Change Humble to use humble branch.

### CODE REVIEW

**Comment by j-rivero** [comment]
looking at CI, me might need to add this new combination.

File: README.md [hunk 1]
```diff
@@ -6,7 +6,8 @@ Foxy | Citadel | [foxy](https://github.com/gazebosim/ros_gz/tree/foxy) | https:/
 Foxy | Edifice | [foxy](https://github.com/gazebosim/ros_gz/tree/foxy) | only from source
 Galactic | Edifice | [galactic](https://github.com/gazebosim/ros_gz/tree/galactic) | https://packages.ros.org
 Galactic | Fortress | [galactic](https://github.com/gazebosim/ros_gz/tree/galactic) | only from source
-Humble | Fortress | [ros2](https://github.com/gazebosim/ros_gz/tree/ros2) | https://packages.ros.org
+Humble | Fortress | [ros2](https://github.com/gazebosim/ros_gz/tree/humble) | https://packages.ros.org
+Humble | Garden (not released) | [ros2](https://github.com/gazebosim/ros_gz/tree/humbre) | only from source
```
Comment (caguero) [inline README.md]: typo: humble

**Comment by j-rivero** [comment]
> looking at CI, me might need to add this new combination.

Testing it [46d16f9](https://github.com/gazebosim/ros_gz/pull/297/commits/46d16f9c0d0b3aa6217ad73cd1fd1ad265e3c87e)

File: README.md (hunk 1 above)
Comment (j-rivero) [reply README.md]: ouk. [fb8aef9](https://github.com/gazebosim/ros_gz/pull/297/commits/fb8aef98ff7a82435a2dff9020727be33e757864)

**Review by caguero — APPROVED** [review]

**Pull request merged**